### PR TITLE
Rejuvenate log levels

### PR DESCRIPTION
**Pull request by saledouble**
We appreciate your previous feedback and it's very helpful! We added Slf4j support and enhanced our tool after that. Here's a reissue of https://github.com/eclipse/californium/pull/933 with a new version of our tool.  Again, we'd appreciate any feedback and are willing to make further changes if you wish to incorporate our PR into your project.

Settings
---
We have several analysis settings. We can vary these settings and rerun if you desire. The settings we are using in this pull request are:

- Never lower the logging level of logging statements within catch blocks.
- Never lower the logging level of logging statements within if statements.
- Never lower the logging level of logging statements containing certain (important) keywords.
- Never raise the logging level of logging statements without particular keywords in their messages.
- Avoid log wrapping by disregarding logging statements contained in if statements mentioning log levels.
- Consistent log level transformations between overriding methods.
- Maximum allowed transformation distance: 2.

The greatest number of commits from HEAD evaluated: 1000.
The head at the time of analysis was: 5b026ab87acd8810b2b8d79ba5ef305c973b8b4e

### CODE REVIEW

**Comment by boaks** [comment]
Thanks!
I have been too busy to have a look at it. But I will spend now some time.

File: californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpointHealthLogger.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void dump(String tag) {
 				log.append(head).append(receivedRejects).append(eol);
 				log.append(head).append(duplicateRequests).append(eol);
 				log.append(head).append(duplicateResponses);
-				LOGGER.debug("{}", log);
+				LOGGER.trace("{}", log);
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpointHealthLogger.java]: Here the debug-level is important.

File: californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java [hunk 2]
```diff
@@ -247,7 +247,7 @@ public static void reset() {
 
 		@Override
 		public void deliverRequest(Exchange exchange) {
-			LOGGER.error("Default endpoint without CoapServer has received a request.");
+			LOGGER.info("Default endpoint without CoapServer has received a request.");
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java]: That's still an error.

File: californium-core/src/main/java/org/eclipse/californium/core/network/RandomTokenGenerator.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public RandomTokenGenerator(final NetworkConfig networkConfig) {
 		// trigger self-seeding of the PRNG, may "take a while"
 		this.rng.nextInt(10);
 		this.tokenSize = networkConfig.getInt(Keys.TOKEN_SIZE_LIMIT, DEFAULT_TOKEN_LENGTH);
-		LOGGER.info("using tokens of {} bytes in length", this.tokenSize);
+		LOGGER.trace("using tokens of {} bytes in length", this.tokenSize);
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/RandomTokenGenerator.java]: The intention of info is to log the effective configuration on start. So I would keep info.

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java [hunk 4]
```diff
@@ -192,12 +192,12 @@ public static LogOnlyLayer getInstance() {
 
 		@Override
 		public void sendRequest(final Exchange exchange, final Request request) {
-			LOGGER.error("No lower layer set for sending request [{}]", request);
+			LOGGER.info("No lower layer set for sending request [{}]", request);
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java]: Still an error

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java [hunk 5]
```diff
@@ -192,12 +192,12 @@ public static LogOnlyLayer getInstance() {
...
 		}
 
 		@Override
 		public void sendResponse(final Exchange exchange, final Response response) {
-			LOGGER.error("No lower layer set for sending response [{}]", response);
+			LOGGER.info("No lower layer set for sending response [{}]", response);
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java]: Still an error

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java [hunk 6]
```diff
@@ -207,12 +207,12 @@ public void sendEmptyMessage(Exchange exchange, EmptyMessage emptyMessage) {
 
 		@Override
 		public void receiveRequest(final Exchange exchange, final Request request) {
-			LOGGER.error("No upper layer set for receiving request [{}]", request);
+			LOGGER.info("No upper layer set for receiving request [{}]", request);
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java]: Still an error

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java [hunk 7]
```diff
@@ -207,12 +207,12 @@ public void sendEmptyMessage(Exchange exchange, EmptyMessage emptyMessage) {
...
 		}
 
 		@Override
 		public void receiveResponse(final Exchange exchange, final Response response) {
-			LOGGER.error("No lower layer set for receiving response [{}]", response);
+			LOGGER.info("No lower layer set for receiving response [{}]", response);
```
Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/AbstractLayer.java]: Still an error

File: californium-core/src/test/java/org/eclipse/californium/core/test/CountingCoapHandler.java [hunk 8]
```diff
@@ -74,7 +74,7 @@ public final void onLoad(CoapResponse response) {
 			}
 			notifyAll();
 		}
-		LOGGER.info("Received {}. Notification: {}", counter, response.advanced());
+		LOGGER.trace("Received {}. Notification: {}", counter, response.advanced());
```
Comment (boaks) [inline californium-core/src/test/java/org/eclipse/californium/core/test/CountingCoapHandler.java]: That's a unit test. The logging is intended to analyze the effect of changes somewhere else. So I would keep info.

File: element-connector/src/test/java/org/eclipse/californium/elements/UDPConnectorTest.java [hunk 9]
```diff
@@ -197,7 +197,7 @@ private void testStopCallsMessageCallbackOnError(final int pending, final int lo
 		EndpointContext context = new UdpEndpointContext(dest);
 
 		for (int loop = 0; loop < loops; ++loop) {
-			LOGGER.info("start/stop: {}/{} loops, {} msgs", loop, loops, pending);
+			LOGGER.severe("start/stop: {}/{} loops, {} msgs", loop, loops, pending);
```
Comment (boaks) [inline element-connector/src/test/java/org/eclipse/californium/elements/UDPConnectorTest.java]: `severe` is not available for slf4j.

**Comment by boaks** [comment]
After checking to proposed changes, I still have my doubts about the approach, to determine the log level from the change history. The most proposed changes I would not apply, for the same reasons as [last year](https://github.com/eclipse/californium/pull/933#issuecomment-487458686)

**Comment by saledouble** [comment]
Thank you for your valuable feedback! We will consider it.

**Pull request closed without merging**